log4j-core/src/main/java/org/apache/logging/log4j/core/impl/Log4jContextFactory.java

```diff
@@ -146,8 +149,12 @@ public LoggerContext getContext(
         if (externalContext != null && ctx.getExternalContext() == null) {
             ctx.setExternalContext(externalContext);
         }
-        if (ctx.getState() == LifeCycle.State.INITIALIZED) {
-            startContext(ctx, classLoader);
+        try {
+            if (ctx.getState() == LifeCycle.State.INITIALIZED) {
+                startContext(ctx, classLoader);
+            }
+        } catch (IllegalStateException ex) {
+            return internalContext;
```

Comment:
@rgoers, there are several points introduced in this PR that swallow exceptions. I would appreciate it if you can reconsider _"Don't we need to communicate anything to the user at this point?"_ and if not, add some comments explaining why it is safe to swallow.

Reply:
I agree on the need to log a `WARN` message or even an `ERROR` message.

In these kind of situations users are really clueless on why their Log4j configuration is not working. Problems concerning multiple `LoggerContext`s are the hardest to debug. Users usually **don't** know that multiple logger contexts are possible (this is _almost_ exclusive to Log4j Core).

Reply:
Previously the error was propagated back to LogManager. It would log a Warn message and then promptly get a ClassCastException as it returned the LoggerContext from logj-api, not a LoggerContext compatible with log4j-core. I think I might have a better way to deal with this so I will look into it.